src/main/java/dk/xam/jbang/Main.java

```diff
@@ -81,259 +79,377 @@
-	Boolean cds;
+	static boolean verbose = false;
 
-	Optional<Boolean> cds() {
-		return Optional.ofNullable(cds);
+	void info(String msg) {
+		spec.commandLine().getErr().println("[jbang] " + msg);
 	}
 
-	private Script script;
+	void warn(String msg) {
+		info("[WARNING] " + msg);
+	}
 
-	boolean debug() {
-		return debugPort >= 0;
+	boolean isVerbose() {
+		return verbose;
 	}
 
-	@Option(names = { "-e", "--edit" }, description = "Edit script by setting up temporary project.")
-	boolean edit;
+	@Override
+	public Integer call() throws IOException {
+		try {
+			return doCall();
+		} catch (ExitException e) {
+			if (isVerbose()) {
+				e.printStackTrace();
+			} else {
+				info(e.getMessage());
+				info("Run with --verbose for more details");
+			}
 
-	@Option(names = {
-			"--edit-live" }, description = "Setup temporary project, launch editor and regenerate project on dependency changes.", parameterConsumer = PlainStringFallbackConsumer.class, arity = "0..1", fallbackValue = "${JBANG_EDITOR:-${VISUAL:-${EDITOR}}}")
-	String liveeditor;
+			return e.getStatus();
+		}
+	}
 
-	@Option(names = { "-o", "--offline" }, description = "Work offline. Fail-fast if dependencies are missing.")
-	boolean offline;
+	public abstract Integer doCall() throws IOException;
+}
 
-	@Option(names = { "-h", "--help" }, usageHelp = true, description = "Display help/info")
-	boolean helpRequested;
+@Command(name = "jbang", footer = "\nCopyright: 2020 Max Rydahl Andersen, License: MIT\nWebsite: https://github.com/jbangdev/jbang", description = "Compiles and runs .java/.jsh scripts.", subcommands = {
+		JbangRun.class, JbangScript.class, JbangAlias.class, JbangTrust.class, JbangClearCache.class,
+		JbangCompletion.class, JbangVersion.class })
+public class Main extends JbangBaseCommand {
 
-	@Option(names = { "--version" }, versionHelp = true, arity = "0", description = "Display version info")
-	boolean versionRequested;
+	@Option(names = { "--verbose" }, description = "jbang will be verbose on what it does.", scope = ScopeType.INHERIT)
+	void setVerbose(boolean verbose) {
+		JbangBaseCommand.verbose = verbose;
+	}
 
-	@Option(names = {
-			"--clear-cache" }, help = true, description = "Clear cache of dependency list and temporary projects")
-	boolean clearCache;
+	public Integer doCall() {
+		spec.commandLine().usage(err);
+		return 0;
+	}
 
-	@Option(names = { "--verbose" }, description = "jbang will be verbose on what it does.")
-	boolean verbose;
+	public static void main(String... args) {
+		CommandLine cli = getCommandLine();
+		args = handleDefaultRun(cli.getCommandSpec(), args);
+		int exitcode = cli.execute(args);
+		System.exit(exitcode);
+	}
 
-	@Option(names = { "--interactive" }, description = "activate interactive mode")
-	boolean interactive;
+	private static String[] handleDefaultRun(CommandSpec spec, String[] args) {
```

Comment:
is this the way to deal with defaulting the run command ? I could imagine this is what is messing with completion handling not working or that this injecting of "run" is not recognized by the completor?

Reply:
I'm not sure, but I can't imagine right now how that could cause problems? The completion system doesn't know about this so it should just generate completion options for what's defined, right? I suspect the problem will remain even if we simply comment this code out.